### ISSUE TITLE
Goal 5.4.: Optimize Transformer

### ISSUE THREAD

**Issue by bschink** [bschink/pocket-narrator#22]
- [x] optimize trainer: Automatic Mixed Precision, Gradient Clipping, Learning Rate Scheduling
- [x] optimize architecture: KV-Caching

**Comment by bschink** [comment]
everything is implemented
